File: perl5/day01.pl
#!/usr/bin/perl
use strict;
use warnings;

open my $file, "<", "../day01.input"
	or die $!;

my $floor = 0;
while(my $line = <$file>) {
	chomp $line;
	for my $char (split //, $line) {
		$floor++ if $char eq "(";
		$floor-- if $char eq ")";
	}
}

print "$floor\n";

File: go/day01.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, err := os.Open("../day01.input")
	if err != nil {
		panic(err)
	}

	floor := 0
	b1 := make([]byte, 1)
	for {
		_, err := f.Read(b1)
		if err != nil {
			break
		}

		if string(b1) == "(" {
			floor++
		} else if string(b1) == ")" {
			floor--
		}
	}

	fmt.Println(floor)
}
